docs/pages/versions/unversioned/sdk/app-integrity.mdx

```diff
@@ -0,0 +1,186 @@
+
+> **important**  Most app extensions don’t support App Attest. Generally, when executing code in these extensions, bypass key generation and attestation, even if the `isSupported` method property is `true`. The only app extensions that support App Attest are watchOS extensions in watchOS 9 or later. For these extensions, you can use the results from `isSupported` to indicate whether your WatchKit extension bypasses attestation.
+
+
+### Create a key pair
+
+For each user account on each device running your app, generate a unique, hardware-based, cryptographic key pair by calling the `generateKey` method.
+
+```js
+const keyId = await AppIntegrity.generateKey()
+```
+
+On success, the method returns a key identifier (`keyId`) that you use later to access the key. Record the identifier in persistent storage because there’s no way to use the key without the identifier, and no way to get the identifier later. The device automatically stores the associated private key in the Secure Enclave, from where the App Attest service can use it to create signatures, but from where no process can ever directly read or modify it, ensuring its security.
+
+> **Important** If you create a key pair in an App Clip, use the same key pair in the corresponding app. To support this, be sure to store the identifier in a shared container accessible from your full app.
+
+Don’t reuse a key among multiple users on a device because this weakens security protections. In particular, it becomes hard to detect an attack that uses a single compromised device to serve multiple remote users running a compromised version of your app. For more information, see [Assessing fraud risk](https://developer.apple.com/documentation/devicecheck/assessing-fraud-risk).
+
+### Get a challenge from your server
+
+Request a unique, one-time challenge from your server. This challenge will be embedded in the attestation step below, ensuring it can’t be reused by an attacker. The challenge should be at least 16 bytes in length to ensure sufficient entropy to ensure guessing them is infeasible.
+
+### Certify the key pairs as valid
+
+Pass the `keyId` alongwith the challenge from your server created in the previous steps in `attestKey` method as shown below:
+
+```js
+const attestationObject = await AppIntegrity.attestKey(keyId, challenge)
+```
+On success, send the received `attestationObject` and the `keyId` to your server for verification.
+
+If the method returns Service unavailable error, try attestation again later with the same key. For any other error, discard the key identifier and create a new key when you want to try again.
+
+> **Important** If your app already has millions of daily active users and you want to start calling the `attestKey` method to initiate attestation from your app, please review [Preparing to use the app attest service](https://developer.apple.com/documentation/DeviceCheck/preparing-to-use-the-app-attest-service) for guidance on safely ramping your users.
```

Comment:
```suggestion
> **info** If your app already has millions of daily active users and you want to start calling the `attestKey` method to initiate attestation from your app, review the [Preparing to use the app attest service](https://developer.apple.com/documentation/DeviceCheck/preparing-to-use-the-app-attest-service) for guidance on safely ramping your users.
```

Let's avoid using words like "please". We need to be informative and instructive through docs.